src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1353,44 +1359,41 @@ private JSONObject getMailArchiveZipContent(Environment env, String site_id,
 					}
 
 				} else if (Utils.isItMailArchiveMbox(destination_type)) {
+					String messageFolderName = getMailArchiveMboxMessageFolderName(site_id);
+					ZipEntry fileEntry = new ZipEntry(messageFolderName + Utils.MAIL_MBOX_MESSAGE_FILE_NAME);
+					out.putNextEntry(fileEntry);
 					for (int iMessage = 0; iMessage < messages.length(); iMessage++) {
 						JSONObject message = messages.getJSONObject(iMessage);
```

Comment:
The for loop should just wrap a call to a method that does the processing for a single method.  That would be much easier to understand and test.